Model/PaymentMethod.php

```diff
@@ -134,8 +134,8 @@ public function __construct(
         \Magento\Directory\Model\RegionFactory $regionFactory,
         \Magento\Sales\Api\OrderRepositoryInterface $orderRepository,
         \Razorpay\Magento\Controller\Payment\Order $order,
-        \Magento\Framework\Model\ResourceModel\AbstractResource $resource = null,
-        \Magento\Framework\Data\Collection\AbstractDb $resourceCollection = null,
+        AbstractResourceMethod $resource = null,
+        AbstractDbMethod $resourceCollection = null,
```

Comment:
@enishant , the new class or model not required here. Please make the optional fields params at the end of function define. It's php8.1 constraints.